presto-main/src/test/java/io/prestosql/sql/analyzer/TestAnalyzer.java

```diff
@@ -1711,45 +1700,45 @@ private void analyze(Session clientSession, @Language("SQL") String query)
                 });
     }
 
-    private void assertFails(SemanticErrorCode error, @Language("SQL") String query)
+    private void assertFails(StandardErrorCode error, @Language("SQL") String query)
     {
         assertFails(CLIENT_SESSION, error, query);
     }
 
-    private void assertFails(SemanticErrorCode error, int line, int column, @Language("SQL") String query)
+    private void assertFails(StandardErrorCode error, int line, int column, @Language("SQL") String query)
     {
-        assertFails(CLIENT_SESSION, error, Optional.of(new NodeLocation(line, column - 1)), query);
+        assertFails(CLIENT_SESSION, error, Optional.of(new Location(line, column)), query);
     }
 
-    private void assertFails(SemanticErrorCode error, String message, @Language("SQL") String query)
+    private void assertFails(StandardErrorCode error, String message, @Language("SQL") String query)
     {
         assertFails(CLIENT_SESSION, error, message, query);
     }
 
-    private void assertFails(Session session, SemanticErrorCode error, @Language("SQL") String query)
+    private void assertFails(Session session, StandardErrorCode error, @Language("SQL") String query)
     {
         assertFails(session, error, Optional.empty(), query);
     }
 
-    private void assertFails(Session session, SemanticErrorCode error, Optional<NodeLocation> location, @Language("SQL") String query)
+    private void assertFails(Session session, StandardErrorCode error, Optional<Location> location, @Language("SQL") String query)
     {
         try {
             analyze(session, query);
             fail(format("Expected error %s, but analysis succeeded", error));
         }
-        catch (SemanticException e) {
-            if (e.getCode() != error) {
-                fail(format("Expected error %s, but found %s: %s", error, e.getCode(), e.getMessage()), e);
+        catch (PrestoException e) {
+            if (e.getErrorCode() != error.toErrorCode()) {
+                fail(format("Expected error %s, but found %s: %s", error, e.getErrorCode().getName(), e.getMessage()), e);
             }
 
             if (location.isPresent()) {
-                NodeLocation expected = location.get();
-                NodeLocation actual = e.getNode().getLocation().get();
+                Location expected = location.get();
+                Location actual = e.getLocation().get();
```

Comment:
This can throw. Change to something like
```java
Location actual = e.getLocation().orElseThrow(() ->
        new AssertionError("Expected location but actual is missing"));